src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -0,0 +1,215 @@
+            currentGroup.setState(employee.getAddressStateProvinceCodeUnmasked());
+            currentGroup.setZip(employee.getAddressPostalCodeUnmasked());
+        }
+    }
+    
+    protected String buildPayeeName(String lastName, String firstName, String middleInitial) {
+        String separator = KFSConstants.COMMA + KFSConstants.BLANK_SPACE;
+        String fullName = lastName + separator + firstName;
+        if (StringUtils.isNotBlank(middleInitial)) {
+            fullName = fullName + KFSConstants.BLANK_SPACE + middleInitial + KFSConstants.DELIMITER;
+        }
+        if(fullName.length() > getPayeeNameFieldSize()) {
+            fullName = StringUtils.substring(fullName, 0, getPayeeNameFieldSize().intValue());
+            fullName = removeLastCharacterWhenComma(fullName);
+        }
+        return fullName;
+    }
+    
+    private String removeLastCharacterWhenComma(String fullName) {
+        if (fullName.lastIndexOf(KFSConstants.COMMA) >= getPayeeNameFieldSize()-2) {
+            fullName = fullName.substring(0, fullName.lastIndexOf(KFSConstants.COMMA));
+        }
+        return fullName;
+    }
+
+    @Override
+    public PdpFeedPayeeIdEntry buildPayeeIdEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedPayeeIdEntry payeeIdEntry = new PdpFeedPayeeIdEntry();
+        payeeIdEntry.setContent(line.getEmployeeId());
+        if (StringUtils.equalsIgnoreCase(line.getEmployeeStatus(), ConcurConstants.EMPLOYEE_STATUS_CODE)) {
+            payeeIdEntry.setIdType(ConcurConstants.EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else if (StringUtils.equalsIgnoreCase(line.getEmployeeStatus(), ConcurConstants.NON_EMPLOYEE_STATUS_CODE)) {
+            payeeIdEntry.setIdType(ConcurConstants.NON_EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else {
+            LOG.error("buildPayeeIdEntry, Unable to to set the payee ID type based do the line's employee status " + line.getEmployeeStatus());
+        }
+        return payeeIdEntry;
+    }
+
+    @Override
+    public PdpFeedDetailEntry buildPdpFeedDetailEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedDetailEntry currentDetailEntry  = new PdpFeedDetailEntry();
+        currentDetailEntry.setSourceDocNbr(buildSourceDocumentNumber(line.getReportId()));
+        currentDetailEntry.setFsOriginCd(getConcurParameterValue(ConcurParameterConstants.CONCUR_AP_PDP_ORIGINATION_CODE));
+        currentDetailEntry.setFdocTypCd(getConcurParameterValue(ConcurParameterConstants.CONCUR_SAE_PDP_DOCUMENT_TYPE));
+        currentDetailEntry.setInvoiceNbr(StringUtils.EMPTY);
+        currentDetailEntry.setPoNbr(StringUtils.EMPTY);
+        currentDetailEntry.setInvoiceDate(formatDate(line.getBatchDate()));
+        List<String> paymentTextStrings = new ArrayList<String>();
+        paymentTextStrings.add("Report ID: " + line.getReportId());
+        currentDetailEntry.setPaymentText(paymentTextStrings);
```

Comment:
I am going to remove this for now, and put it back in after I nail down the requirements from Marcia and crew